pkg/asset/rhcos/image.go

```diff
@@ -141,13 +141,32 @@ func osImage(config *types.InstallConfig) (string, error) {
 			return oi, nil
 		}
 
-		// Note that baremetal IPI currently uses the OpenStack image
-		// because this contains the necessary ironic config drive
-		// ignition support, which isn't enabled in the UPI BM images
-		if a, ok := streamArch.Artifacts["openstack"]; ok {
-			return rhcos.FindArtifactURL(a)
+		// Note that baremetal IPI currently uses the RHCOS live-iso image.
+		// We will need to pass 4 URLs one for virtual media and 3 for PXE
+		// booting. We will concatenate the URLs with a separator for easier extraction.
+		// Note: We cannot use rhcos.FindArtifactURL for the "metal" artifact.
+		iso := streamArch.Artifacts["metal"].Formats["iso"].Disk.Location
+		if iso == "" {
+			return "", fmt.Errorf("%s: No live-iso build found", st.FormatPrefix(archName))
```

Comment:
Based on the today's conversation, let's delay making the ISO mandatory. It's actually not, we can leave happily with only the PXE artefacts.

Reply:
+1. Fixed in the next commit.